### PR TITLE
Update Datastore and Messages to accommodate for replies v2

### PR DESCRIPTION
**Pull request by amuletspade231**

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -62,14 +50,20 @@ public void storeMessage(Message message) {
       try {
         String idString = entity.getKey().getName();
         UUID id = UUID.fromString(idString);
+        UUID parent = UUID.fromString(idString);
```
Comment (dj-smith) [inline src/main/java/com/google/codeu/data/Datastore.java]: So for now, the parent of a comment is just itself? Or am I misunderstanding?

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -111,7 +131,7 @@ public User getUser(String email) {
    }
 
   /**
-   * Gets messages posted by a user, or all messages if user is null.
+   * Gets messages posted to/by a user, or all messages if user is null.
    *
    * @return a list of any messages posted by the user, sorted by time descending. If user is null, returns all messages in the Datastore.
```
Comment (dj-smith) [inline src/main/java/com/google/codeu/data/Datastore.java]: Nitpick -- if the comment is changed to reflect the new setup, the return value description probably should be too

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -129,6 +149,25 @@ public User getUser(String email) {
     return messages;
   }
 
+ /**
+  * Gets replies of a message, or all messages if parent is null.
+  *
+  * @return a list of any replies to a certain parent message, sorted by time descending. If user is null, returns all replies in the Datastore.
+  */
+  public List<Message> getReplies(String parent) {
```
Comment (dj-smith) [inline src/main/java/com/google/codeu/data/Datastore.java]: This nearly identical to getAllMessages. I wonder if there's a way to consolidate them without compromising readability? I wouldn't worry about it right now in terms of prioritization, just wanted to mention the possibility.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (amuletspade231) [reply src/main/java/com/google/codeu/data/Datastore.java]: My bad! It should get the parent property from the entity.

**Pull request merged**